### PR TITLE
CEDS-1484 Add Submission & Notifications endpoint

### PR DESCRIPTION
**Pull request by ed0906**

### CODE REVIEW

File: conf/app.routes [hunk 1]
```diff
@@ -1,10 +1,12 @@
...
-DELETE      /v2/declarations/:id                              uk.gov.hmrc.exports.controllers.DeclarationController.deleteByID(id: String)
+POST        /v2/declarations                                 uk.gov.hmrc.exports.controllers.DeclarationController.create()
+GET         /v2/declarations                                 uk.gov.hmrc.exports.controllers.DeclarationController.findAll(status: Option[String] ?= None, page: uk.gov.hmrc.exports.models.Page ?= uk.gov.hmrc.exports.models.Page())
+PUT         /v2/declarations/:id                             uk.gov.hmrc.exports.controllers.DeclarationController.update(id: String)
+GET         /v2/declarations/:id                             uk.gov.hmrc.exports.controllers.DeclarationController.findByID(id: String)
+DELETE      /v2/declarations/:id                             uk.gov.hmrc.exports.controllers.DeclarationController.deleteByID(id: String)
+GET         /v2/declarations/:id/submission                  uk.gov.hmrc.exports.controllers.SubmissionController.findByID(id: String)
+GET         /v2/declarations/:id/submission/notifications    uk.gov.hmrc.exports.controllers.NotificationController.findByID(id: String)
```
Comment (ed0906) [inline conf/app.routes]: These should not allow us to get rid of the dependency on Conversation ID

File: app/uk/gov/hmrc/exports/controllers/JSONResponses.scala [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.exports.controllers
+
+import play.api.http.{ContentTypeOf, ContentTypes, Writeable}
+import play.api.libs.json.Writes
+import play.api.mvc.Codec
+
+trait JSONResponses {
```
Comment (PatrykRudnicki) [inline app/uk/gov/hmrc/exports/controllers/JSONResponses.scala]: I think putting this in trait and extending every Controller with this it's not a good idea, but it's just my opinion.

File: app/uk/gov/hmrc/exports/controllers/NotificationController.scala [hunk 3]
```diff
@@ -42,17 +41,27 @@ class NotificationController @Inject()(
...
+          .map(notifications => Ok(notifications))
+      case _ => Future.successful(NotFound)
+    }
+  }
+
   def getSubmissionNotifications(mrn: String): Action[AnyContent] =
     authorisedAction(bodyParsers.default) { implicit request =>
       notificationsService
         .getNotificationsForSubmission(mrn)
-        .map(notifications => Ok(Json.toJson(notifications)))
```
Comment (PatrykRudnicki) [inline app/uk/gov/hmrc/exports/controllers/NotificationController.scala]: And I prefer previous explicit `Json.toJson` I'm not sure that you're solving some problem or it's just improvement to write less code

Comment (ed0906) [reply app/uk/gov/hmrc/exports/controllers/NotificationController.scala]: Personally if I don't have to write `Json.toJson` each time its better

Comment (PatrykRudnicki) [reply app/uk/gov/hmrc/exports/controllers/NotificationController.scala]: Yeah, but for new Controller you need to remember about this Trait, I'm not sure about it, let's wait for other people :) I don't want to block PR :)

Comment (ed0906) [reply app/uk/gov/hmrc/exports/controllers/NotificationController.scala]: New controllers will extend the RESTController trait which pulls it in for free. I didnt want to change this one as it's likely to be removed once we stop using conversation-id everywhere

Comment (mattclark-zerogravit) [reply app/uk/gov/hmrc/exports/controllers/NotificationController.scala]: I agree with Patryk, its not obvious you need to include this trait. Also the Json.toJson whilst more verbose shows there is a conversion actually happening. We are invoking the Json writer here. Removing it hides that and it all just happens magically which reads nicer but could argue is more confusing.

Comment (ed0906) [reply app/uk/gov/hmrc/exports/controllers/NotificationController.scala]: For me less boiler plate is better, Play is the only framework I've seen that makes you manually convert all payloads to & from strings, others e.g. SpringBoot, DropWizard presume you want JSON and do it for you.
 
Regarding creating new controllers, it's easy to miss a lot of things so we'd generally copy and paste an existing one anyway. Any new Controllers would extend `RESTController` which gives everything for free

Comment (SteveSugden) [reply app/uk/gov/hmrc/exports/controllers/NotificationController.scala]: If the DeclarationController (which extends RESTController) is the accepted "new" way of creating json responses then I see no reason why the Notification and Submission controllers shouldn't (temporarily?) do the same thing.

**Pull request merged**